GVRf/Framework/src/org/gearvrf/scene_objects/GVRCubeSceneObject.java

```diff
@@ -398,4 +436,415 @@ private void createSimpleCubeSixMeshes(GVRContext gvrContext,
+                subMeshes[index].setVertices(vertices);
+                subMeshes[index].setNormals(normals);
+                subMeshes[index].setTexCoords(texCoords);
+                subMeshes[index].setTriangles(indices);
+                grandchildren[index] = new GVRSceneObject(gvrContext,
+                        new FutureWrapper<GVRMesh>(subMeshes[index]),
+                        futureTextureList.get(4));
+                children[4].addChildObject(grandchildren[index]);
+            }
+        }
+
+        // bottom face
+        normals[0] = normals[3] = normals[6] = normals[9] = 0.0f;
+        if (facingOut) {
+            normals[1] = normals[4] = normals[7] = normals[10] = -1.0f;
+        } else {
+            normals[1] = normals[4] = normals[7] = normals[10] = 1.0f;
+        }
+        normals[2] = normals[5] = normals[8] = normals[11] = 0.0f;
+        for (int col = 0; col<segmentNumber; col++) {
+            for (int row = 0; row<segmentNumber; row++) {
+                // sub-mesh (col, row)
+                int index = row*segmentNumber+col;
+                
+                float y = -SIZE;
+                float x0 = -SIZE + segmentLength * col;
+                float z0 = -SIZE + segmentLength * row;
+                float x1 = x0 + segmentLength;
+                float z1 = z0 + segmentLength;
+                vertices[0] = x0;
+                vertices[1] = y;
+                vertices[2] = z0;
+                vertices[3] = x1;
+                vertices[4] = y;
+                vertices[5] = z0;
+                vertices[6] = x0;
+                vertices[7] = y;
+                vertices[8] = z1;
+                vertices[9] = x1;
+                vertices[10] = y;
+                vertices[11] = z1;
+                
+                float s0, s1;
+                s0 = col * segmentTexCoordLength;
+                s1 = (col + 1) * segmentTexCoordLength;
+                float t0 = 1.0f - (row + 1) * segmentTexCoordLength;
+                float t1 = 1.0f - row * segmentTexCoordLength;
+                texCoords[0] = s0; 
+                texCoords[1] = t1; 
+                texCoords[2] = s1; 
+                texCoords[3] = t1; 
+                texCoords[4] = s0; 
+                texCoords[5] = t0; 
+                texCoords[6] = s1; 
+                texCoords[7] = t0;
+                
+                subMeshes[index] = new GVRMesh(gvrContext);
+                subMeshes[index].setVertices(vertices);
+                subMeshes[index].setNormals(normals);
+                subMeshes[index].setTexCoords(texCoords);
+                subMeshes[index].setTriangles(indices);
+                grandchildren[index] = new GVRSceneObject(gvrContext,
+                        new FutureWrapper<GVRMesh>(subMeshes[index]),
+                        futureTextureList.get(5));
+                children[5].addChildObject(grandchildren[index]);
+            }
+        }
+
```

Comment:
Still the same as before, the level of code duplicate is a little bit high. We may find some way to have a better code structure that can reuse the codes well.